porch/engine/pkg/engine/engine.go

```diff
@@ -102,6 +115,14 @@ func (cad *cadEngine) CreatePackageRevision(ctx context.Context, repositoryObj *
 
 func (cad *cadEngine) mapTaskToMutation(ctx context.Context, obj *api.PackageRevision, task *api.Task) (mutation, error) {
 	switch task.Type {
+	case api.TaskTypeInit:
```

Comment:
I do wonder if an init could be a clone with no upstream, but you're right that this is likely clearer!

Reply:
We could also do clone from a 'singleton' (similar to what Docker does with `FROM scratch`.

The growing problem is representing actions as tasks, especially as Porch does more things automatically, things that there are no tasks for.